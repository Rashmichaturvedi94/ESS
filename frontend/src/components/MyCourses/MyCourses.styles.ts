import styled from 'styled-components';
import { MyCoursesProps } from './MyCourses.interface';

export const MyCourses = styled.div<Pick<MyCoursesProps, 'myProp'>>`
  /* Add styles here */
`;
