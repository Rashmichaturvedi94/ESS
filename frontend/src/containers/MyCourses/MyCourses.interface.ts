export interface MyCoursesProps {
  myProp?: boolean;
  /* Add props here. */
}
