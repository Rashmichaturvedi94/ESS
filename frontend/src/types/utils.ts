export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
