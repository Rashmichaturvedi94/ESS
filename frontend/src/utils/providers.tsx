import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
