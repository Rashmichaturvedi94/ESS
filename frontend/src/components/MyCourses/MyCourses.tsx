import React, { FC } from 'react';
import { MyCourses as MyCoursesComponent } from './MyCourses.styles';
import { MyCoursesProps } from './MyCourses.interface';

export const MyCourses: FC<MyCoursesProps> = ({ children }) => (
  <MyCoursesComponent>{children}</MyCoursesComponent>
);
