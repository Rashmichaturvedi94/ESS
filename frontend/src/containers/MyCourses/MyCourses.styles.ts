import styled from 'styled-components';
import { Typography } from '@material-ui/core';
import { MyCoursesProps } from './MyCourses.interface';

export const MyCourses = styled.div<Pick<MyCoursesProps, 'myProp'>>`
  /* Add styles here */
`;

export const SubscribedTitle = styled(Typography)`
  border-bottom: 1px;
  border-color: black;
`;
