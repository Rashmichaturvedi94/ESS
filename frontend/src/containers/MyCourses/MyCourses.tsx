import React, { FC } from "react";
import { Box, GridList, Typography, GridListTile, Divider } from "@material-ui/core";
import { useHistory, generatePath  } from 'react-router-dom';
import { appPaths } from '../../const/paths';
import { MyCoursesProps } from "./MyCourses.interface";
import { useSubscriptions } from "../../api";
import { Course } from "../../api/api.interface";

export const MyCourses: FC<MyCoursesProps> = () => {
  const subscriptions = useSubscriptions();
  const history = useHistory();

  const getImage = (course: Course) =>
    typeof course.img === "string" ? course.img : "";
  return (
    <Box style={{ marginLeft: 20, marginRight: 20 }}>
      <Typography variant="h4" component="h2" align="left">
        Subscribed
      </Typography>
      <Divider style={{ color: "black", height: 5, marginTop: 30, marginBottom: 30 }} />
      <GridList>
        {subscriptions.data?.map((subscription) => (
          <GridListTile
            key={subscription.course.id}
            onClick={() => history.push(generatePath(appPaths.courseContent, { courseId: subscription.course.id }))}
            style={{
              height: 258,
              width: 224,
              border: "2px solid black",
              margin: 30,
            }}
          >
            <img
              src={getImage(subscription.course)}
              style={{ height: 196, width: 220, overflow: "hidden" }}
              alt={subscription.course.title}
            />
            <Typography>{subscription.course.title}</Typography>
            <Typography>{subscription.course.description}</Typography>
          </GridListTile>
        ))}
      </GridList>
    </Box>
  );
};
