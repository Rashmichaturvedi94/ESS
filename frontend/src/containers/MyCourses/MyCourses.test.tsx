import React from 'react';
import { render } from '../../utils/testUtils';
import { MyCourses } from './MyCourses';

describe('components/MyCourses', () => {
  it('should render', () => {
    const { container } = render(<MyCourses />);
    expect(container).toMatchSnapshot();
  });
});
