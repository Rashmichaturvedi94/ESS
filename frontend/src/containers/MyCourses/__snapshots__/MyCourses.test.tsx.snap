// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MyCourses should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
    style="margin-left: 20px; margin-right: 20px;"
  >
    <h2
      class="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft"
    >
      Subscribed
    </h2>
    <hr
      class="MuiDivider-root"
      style="color: black; height: 5px; margin-top: 30px; margin-bottom: 30px;"
    />
    <ul
      class="MuiGridList-root"
      style="margin: -2px;"
    />
  </div>
</div>
`;
